File: LeetCode/cpp/1046.cpp
class Solution {
public:
    int lastStoneWeight(vector<int> &stones) {
        priority_queue<int> heap;
        for (auto &stone:stones)
            heap.push(stone);
        while (!heap.empty()) {
            int i = heap.top();
            heap.pop();
            if (heap.empty())
                return i;
            int j = heap.top();
            heap.pop();
            heap.push(i - j);
        }
        return heap.top();
    }
};

File: LeetCode/python/1046.py
class Solution(object):
    def lastStoneWeight(self, stones):
        """
        :type stones: List[int]
        :rtype: int
        """
        h = [-s for s in stones]
        heapq.heapify(h)
        while h:
            i = -heapq.heappop(h)
            if not h:
                return i
            i += heapq.heappop(h)
            if i != 0:
                heapq.heappush(h, -i)
        return -heapq.heappop(h) if h else 0
